tests/Backend/ExternalBuckets/BigqueryRegisterBucketTest.php

```diff
@@ -237,6 +237,124 @@ public function testRefreshBucketWhenSchemaDoesNotExist(): void
+        $this->assertSame(['AMOUNT', 'DESCRIPTION'], $tableDataPreview['columns']);
+        $this->assertSame(['AMOUNT', 'DESCRIPTION'], $viewDataPreview1['columns']);
+
+        $this->_client->refreshBucket($idOfBucket);
+
+        // test after refresh, still works
+        /** @var array $tableDataPreview */
+        $tableDataPreview = $this->_client->getTableDataPreview($table['id'], ['format' => 'json']);
+        /** @var array $viewDataPreview1 */
+        $viewDataPreview1 = $this->_client->getTableDataPreview($view['id'], ['format' => 'json']);
+        $this->assertSame(['AMOUNT', 'DESCRIPTION', 'AGE'], $tableDataPreview['columns']);
+        $this->assertSame(['AMOUNT', 'DESCRIPTION', 'AGE'], $viewDataPreview1['columns']);
+
+        // drop column and recreate view, to test preview ends with user err
+        $db->executeQuery(sprintf(
+        /** @lang BigQuery */
+            'ALTER TABLE `%s`.`MY_LITTLE_TABLE_FOR_VIEW`  DROP COLUMN AGE;',
+            $schemaName,
+        ));
+
+        $db->executeQuery(sprintf(
+        /** @lang BigQuery */
+            'CREATE OR REPLACE VIEW `%s`.`MY_LITTLE_VIEW` AS SELECT * FROM `%s`.`MY_LITTLE_TABLE_FOR_VIEW`',
+            $schemaName,
+            $schemaName,
+        ));
+
+        try {
+            $this->_client->getTableDataPreview($table['id'], ['format' => 'json']);
+            $this->fail('Should fail');
+        } catch (ClientException $e) {
+            $this->assertSame(400, $e->getCode());
+            $this->assertSame('storage.backend.externalBucketObjectInvalidIdentifier', $e->getStringCode());
+            $this->assertStringContainsString('External object might be out of sync. Please refresh the external bucket to ensure it\'s synchronized, then try again.', $e->getMessage());
+        }
+
+        try {
+            $this->_client->getTableDataPreview($view['id'], ['format' => 'json']);
+            $this->fail('Should fail');
+        } catch (ClientException $e) {
+            $this->assertSame(400, $e->getCode());
+            $this->assertSame('storage.backend.externalBucketObjectInvalidIdentifier', $e->getStringCode());
+            $this->assertStringContainsString('External object might be out of sync. Please refresh the external bucket to ensure it\'s synchronized, then try again.', $e->getMessage());
+        }
+
+        // after refresh should work again
+        $this->_client->refreshBucket($idOfBucket);
+
+        /** @var array $tableDataPreview */
+        $tableDataPreview = $this->_client->getTableDataPreview($table['id'], ['format' => 'json']);
+        /** @var array $viewDataPreview1 */
+        $viewDataPreview1 = $this->_client->getTableDataPreview($view['id'], ['format' => 'json']);
+        $this->assertSame(['AMOUNT', 'DESCRIPTION'], $tableDataPreview['columns']);
+        $this->assertSame(['AMOUNT', 'DESCRIPTION'], $viewDataPreview1['columns']);
```

Comment:
sorry myslel som ze som to tam poslal ale este som robil lepsiu validaciu tak ako to je u snowflaku